packages/terra-tag/src/Tag.jsx

```diff
@@ -66,6 +74,16 @@ class Tag extends React.Component {
     }
   }
 
+  handleClick(event) {
+    if (this.props.onTagClick) {
+      this.props.onTagClick(event.target.id);
```

Comment:
We should not be using event.target.  You are not guaranteed to get the item that you desire.  It might be working because your component is simple.  Since we are using React, the callback should be able to provide the necessary information without trying to rely on event.target.

Reply:
Updated. Thanks